bin/docker-dev-shell

```diff
@@ -115,6 +115,7 @@ CODE_DIR="/home/dependabot/dependabot-core"
 touch .core-bash_history
 mkdir -p dry-run tmp
 docker run --rm -ti \
+  -v "$HOME/.aws:$CODE_DIR/../.aws" \
```

Comment:
We'll eventually need to ensure this is plumbed through over in `dependabot-script` as well if it's not already, but we can worry about that later...

Reply:
I think `bin/docker-dev-shell` should behave as closely as possible as "the service", and this will introduce non-deterministic behavior in the dev container, depending on whether the developer has AWS credentials configured or not, right? I feel this could lead to confusing behavior/weird spec failures, on certain developer machines, which defeats the point of a developer container a bit. Is this necessary for the feature implemented by this PR?

Reply:
Hmm... good point. 

It's not necessary for `docker-dev-shell`, and you're right that's currently primarily intended for dev work, not prod, so let me pull this out.

I do think we should still plumb it through for `dependabot-script` style stuff where it's intended to be run on prod.

But anyone using `docker-dev-shell` as the entrypoint for prod is doing it wrong IMO.

Reply:
https://github.com/dependabot/dependabot-core/pull/6633

Reply:
Agreed on plumbing this for `dependabot-script`, yeah!